helm/cluster-azure/templates/_azure_cluster.tpl

```diff
@@ -21,14 +21,54 @@ spec:
         - {{ .Values.connectivity.network.controlPlane.cidr }}
       - name: node-subnet
         natGateway:
-          name: node-natgateway
+          name: {{ include "resource.default.name" $ }}-node-natgateway
```

Comment:
do we need this ? unless we have more than one cluster in a resourceGroup i think we can stick to the old name 

this would probably delete and recreate node natgateway in all clusters

Reply:
We use to name most of the resources (in vintage) with cluster ID in the prefix (it wasn't very consistent though).

Personally I find it useful if some use cases, like when using Azure portal (easier to see what you're looking at, where you are, e.g. Azure home page is useless for resources without cluster ID in the prefix, because you just see the resource name), or when analyzing and searching in logs, as it's easier to get around, debug, discover and find stuff.

Reply:
> this would probably delete and recreate node natgateway in all clusters

Regarding this, TBH I stopped worrying about situations like this (at least during this initial heavy development alpha stage of the project), unless it is destructive and it breaks something that we really don't want to break even at this stage (e.g. because it creates too much additional work).

We do have a general issue here regarding handling breaking changes in Helm templates, because it harder to deal with those here (in vintage it was easier as operators would reconcile the change in the background), since we have heavy WIP config that is wired into heavy WIP CRs and that config is then exposed publicly. If you add to that that some fields in some CRs are even immutable, it gets quite challenging.

IMO current version `v0.0.15` is also setting expectations, stuff will break, stuff will get destroyed and get recreated if needed, all of which I see as normal during this current development phase. Once we move to `v0.X.0` it should be more stable, and after that fully stabilizing `cluster-azure` (and other `cluster-xxx` apps) should happen before `v1.0.0` at which point breaking changes should not be done at all.

Reply:
Reversed this change for the sake of minimizing changes in the PR. We can talk about naming afterwards.